gtsfm/utils/io.py

```diff
@@ -450,13 +457,49 @@ def read_points_txt(fpath: str) -> Tuple[Optional[np.ndarray], Optional[np.ndarr
     return point_cloud, rgb
 
 
-def read_scene(
-    images_fpath: str, cameras_fpath: str, points_fpath: str
+def read_scene_data_from_colmap_format(
+    data_dir: str
 ) -> Tuple[List[Pose3], List[str], List[Cal3Bundler], np.ndarray, np.ndarray]:
-    """Reads in full scene reconstruction model."""
-    wTi_list, img_fnames = read_images_txt(images_fpath)
-    calibrations = read_cameras_txt(cameras_fpath)
-    point_cloud, rgb = read_points_txt(points_fpath)
+    """Reads in full scene reconstruction model from scene data stored in the COLMAP file format.
+
+    Reference: https://colmap.github.io/format.html
+
+    Args:
+        data_dir: This directory should contain 3 files: either `cameras.txt`, `images.txt`, and `points3D.txt`, or
+            `cameras.bin`, `images.bin`, and `points3D.bin`.
+
+    Returns:
+        5-tuple of:
+            wTi_list: List of camera poses for each image.
+            img_fnames: List of image file names, for each image.
+            calibrations: Calibration object for each camera.
+            point_cloud: Float array of shape (N,3) representing per-point x/y/z coordinates.
+            rgb: Uint8 array of shape (N,3) representing per-point colors.
+    """
+    # Determine whether scene data is stored in a text (txt) or binary (bin) file format.
+    if Path(data_dir, "images.txt").exists():
+        file_format = "txt"
+    elif Path(data_dir, "images.bin").exists():
+        file_format = "bin"
+    else:
+        raise ValueError(
+            f"Unknown file format, as neither `{data_dir}/images.txt` or `{data_dir}/images.bin` could be found."
+        )
+
+    if file_format == "txt":
+        points_fpath = f"{data_dir}/points3D.txt"
+        images_fpath = f"{data_dir}/images.txt"
+        cameras_fpath = f"{data_dir}/cameras.txt"
+        wTi_list, img_fnames = read_images_txt(images_fpath)
+        calibrations = read_cameras_txt(cameras_fpath)
+        point_cloud, rgb = read_points_txt(points_fpath)
+
+    elif file_format == "bin":
+        cameras, images, points3d = colmap_io.read_model(path=data_dir, ext=".bin")
```

Comment:
could we use this for .txt too and do the `colmap2gtsfm` ? if not, maybe add a comment/TODO why?

Reply:
Good point, I'll add a TODO in a follow-up.